feat: Implement Summary Input UI for Affective Annotation Project

teddy-f-47 opened the pull request:
The mechanism for writing tags and impressions in Summary mode is based on the seq2seq task. However, it is not yet connected to the BE; the data is currently only temporarily written in a variable in FE.

Demo:
https://user-images.githubusercontent.com/64476430/192607392-8ba6a6b6-49b4-45c4-a284-335ef4e0c16f.mp4

Screenshots:
<img width="960" alt="ss1" src="https://user-images.githubusercontent.com/64476430/192607468-4730c6ab-231a-4dbd-9f3e-a9924dd6d1c8.PNG">
<img width="960" alt="ss2" src="https://user-images.githubusercontent.com/64476430/192607638-b860bb2d-7e89-4893-9490-2060ed10b35f.PNG">

What's new:
 - frontend/components/tasks/affectiveAnnotation/inputs/TextfieldWithSeq2Seq.vue : Create base component that combines a clickable textfield with a dialog containing the seq2seqbox
 - frontend/components/tasks/affectiveAnnotation/summary/SummaryInput.vue: Create component for inputting Summary data, which consist of `tags` and `impressions`

What's changed:
frontend/pages/projects/_id/article-annotation/index.vue : Add the SummaryInput component into the annotation view page

To do:
 - Move UI strings into the `i18n` directory after the polish language directory has been created
 - Make translations for the UI strings after the polish language directory has been created
 - Connect inputs to BE after the database has been adjusted

artikandri commented on frontend/components/tasks/affectiveAnnotation/inputs/TextfieldWithSeq2Seq.vue:
```diff
@@ -0,0 +1,132 @@
+
+  watch: {
+    answers() {
+      const res = []
```
I think this can be simplified by using the `map` function, what do you think?

teddy-f-47 replied:
Okay, I'm not really used to the map function so my mind keeps thinking with loop. 😅 I will change it

artikandri commented on frontend/components/tasks/affectiveAnnotation/summary/SummaryInput.vue:
```diff
@@ -0,0 +1,134 @@
+
+  methods: {
+    validateText(field, value) {
+      let flagOk = true
```
From what I understand, this flag is used to mark whether a word is already registered or not, so how about renaming the variable to something else, for example: `isWordRegistered` or something like that?

artikandri commented on frontend/components/tasks/affectiveAnnotation/summary/SummaryInput.vue:
```diff
@@ -0,0 +1,134 @@
+      for (let i = 0; i < arrayToCheck.length; i++) {
+        if (arrayToCheck[i].text === value) {
+          flagOk = false
+          alert("Słowo zostało już napisane.")
```
This is just a question: I dont think there is any pattern in doccano that has used the `alert` function to notify users so far, or is there?

teddy-f-47 replied:
I'm not totally sure, but I think no there isn't XD. In that case, probably it is better to write the warning message somewhere inside the dialog, right? Like under the dialog's title maybe XD I was thinking about this too

artikandri replied:
Yes, I think thats a better way to do this XD

artikandri commented on frontend/components/tasks/affectiveAnnotation/summary/SummaryInput.vue:
```diff
@@ -0,0 +1,134 @@
+      }
+    },
+    addImpression(text) {
+      const passingValidation = this.validateText("impressions", text)
```
This is probably my personal preference, but I prefer to name boolean variables with `is` or `has` as prefix since I think its cleared. Maybe rename this variable to "isValidated" or something like that? XD

artikandri commented on frontend/components/tasks/affectiveAnnotation/summary/SummaryInput.vue:
```diff
@@ -0,0 +1,134 @@
+    },
+    updateImpression(annotationId, text) {
+      if (text.length > 0) {
+        const passingValidation = this.validateText("impressions", text)
```
same case

artikandri commented on frontend/pages/projects/_id/article-annotation/index.vue:
```diff
@@ -408,6 +425,45 @@ export default {
+      this.affectiveSummaryTags.push(item)
+    },
+    removeImpression(annotationId) {
+      for (let i = 0; i < this.affectiveSummaryImpressions.length; i++) {
```
I think this can be simplified by using the `Array.find` function, what do you think?

teddy-f-47 replied:
Most probably yes, I will change it too XD

artikandri reviewed (changes requested):
Thank you for the good work! I have tested it and it works well; however I have something to point out in the code. 

As for the polish translation, I will make the PR later this evening. You may choose to update this PR later to adjust the translation or making a new PR.

teddy-f-47 commented:
I have refactored the points mentioned above and I have also moved the warning message from plain js alert into the dialog.

Screenshots:
<img width="960" alt="ss1" src="https://user-images.githubusercontent.com/64476430/192833612-34832681-6e40-4d6b-802c-057aa5855337.png">
<img width="960" alt="ss2" src="https://user-images.githubusercontent.com/64476430/192833615-ca2d8459-7b16-4ed8-906c-d8ef7f232d37.png">
<img width="960" alt="ss3" src="https://user-images.githubusercontent.com/64476430/192833621-f41cbdba-3a25-4e64-b7e3-eee041ab9b51.png">

artikandri reviewed (approved):
I think everything looks good now, good job! Later we might need to move the component to its proper place (the index.vue for the affective annotation project), but it may be done later after i finished the project creation. For now, this may be merged, and you may also adjust the translation. Thank you!

teddy-f-47 commented:
> I think everything looks good now, good job! Later we might need to move the component to its proper place (the index.vue for the affective annotation project), but it may be done later after i finished the project creation. For now, this may be merged, and you may also adjust the translation. Thank you!

Thank you! I think I will wait for the polish translation to be merged, and I will move the UI strings first, so that everything will be more organized XD

haanh764 reviewed (approved):
Thank you for your great work! I have tested and everything are oke now! Approved for merging!

teddy-f-47 commented:
Rebased to include the polish translation directory, moved UI strings to `frontend\i18n\pl\projects\annotation.js`, added translations in other languages.
